examples/chef/common/chef-rvc-mode-delegate.cpp

```diff
@@ -106,17 +108,68 @@ void RvcRunMode::Shutdown()
     }
 }
 
+chip::Protocols::InteractionModel::Status chefRvcRunModeWriteCallback(chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                                                      const EmberAfAttributeMetadata * attributeMetadata,
+                                                                      uint8_t * buffer)
+{
+    // this cluster is only enabled for endpoint 1
+    if (endpointId != 1)
+    {
+        return chip::Protocols::InteractionModel::Status::UnsupportedEndpoint;
+    }
+
+    chip::Protocols::InteractionModel::Status ret;
+    VerifyOrDie(gRvcRunModeInstance != nullptr);
+
+    ModeBase::Instance * clusterInstance = gRvcRunModeInstance;
+    chip::AttributeId attributeId        = attributeMetadata->attributeId;
+
+    switch (attributeId)
+    {
+    case chip::app::Clusters::RvcRunMode::Attributes::CurrentMode::Id: {
+        uint8_t m = static_cast<uint8_t>(buffer[0]);
+        ret       = clusterInstance->UpdateCurrentMode(m);
+        if (chip::Protocols::InteractionModel::Status::Success != ret)
+        {
+            ChipLogError(DeviceLayer, "Invalid Attribute Update status: %d", static_cast<int>(ret));
+            break;
+        }
+    }
+    break;
+    default:
+        ret = chip::Protocols::InteractionModel::Status::UnsupportedAttribute;
+        ChipLogError(DeviceLayer, "Unsupported Attribute ID: %d", static_cast<int>(attributeId));
+        break;
+    }
+
+    return ret;
+}
+
+chip::Protocols::InteractionModel::Status chefRvcRunModeReadCallback(chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                                                     const EmberAfAttributeMetadata * attributeMetadata,
+                                                                     uint8_t * buffer, uint16_t maxReadLength)
+{
+    if (sizeof(*attributeMetadata) <= sizeof(buffer))
+    {
+        memcpy(buffer, &attributeMetadata, sizeof(attributeMetadata));
```

Comment:
why is this copying the metadata into the buffer? 

I do not quite understand this code. I thought the intent of `read` is to read the specified attribute (and given ember, the attribute would be in ember format ... like raw bytes for integers, pascal strings for strings and binary data).

Why is copying the metadata into the buffer the correct way to handle a read?